model/task_lifecycle.go

```diff
@@ -1193,8 +1198,8 @@ func UpdateDisplayTask(t *task.Task) error {
 
 	sort.Sort(task.ByPriority(execTasks))
 	statusTask = execTasks[0]
-	if hasFinishedTasks && hasUnstartedTasks {
-		// if the display task has a mix of finished and unfinished tasks, the display task is still
+	if !hasBlockedtasks && hasFinishedTasks && hasUnstartedTasks {
+		// if an unblocked display task has a mix of finished and unfinished tasks, the display task is still
```

Comment:
I think this shouldn't consider blocked tasks, because even if some tasks are blocked, some are still dispatchable so we'd still consider this display task in progress (before hasUnstartedTasks included blocked tasks but now it does not; this might be better labeled as `hasDispatchableTasks`)

Reply:
ah good catch. I changed the meaning of hasUnstartedTasks to hasDispatchableTasks and added a test for this case